drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c

```diff
@@ -672,6 +690,14 @@ static void stop_streaming(struct vb2_queue *vq)
 		}
 	}
 
+	/* Release the VCHIQ buffers */
+	for (i = 0; i < vq->num_buffers; i++) {
+		vb2 = to_vb2_v4l2_buffer(vq->bufs[i]);
+		buf = container_of(vb2, struct vb2_mmal_buffer, vb);
+
+		mmal_vchi_buffer_cleanup(&buf->mmal);
+	}
```

Comment:
Why has this loop suddenly become necessary?

Reply:
As above, the vchi context is now attached to the buffer at allocation, not allocated on demand. I'll review to check that bits got added at the correct point.